frame/treasury/src/lib.rs

```diff
@@ -535,12 +559,15 @@ decl_module! {
 		///   as the hash of the tuple of the original tip `reason` and the beneficiary account ID.
 		///
 		/// # <weight>
-		/// - `O(T)`
-		/// - One storage retrieval (codec `O(T)`) and two removals.
-		/// - Up to three balance operations.
+		/// - Complexity: `O(T)` where `T` is the number of tippers.
+		///   decoding `Tipper` vec of length `T`.
+		///   `T` is less than `MAX_TIPPERS_COUNT`, it is charged as maximum and then refund.
+		///   The actual cost depends on the implementation of `T::Tippers`.
+		/// - DbReads: `Tips`, `Tippers`, `tip finder`
+		/// - DbWrites: `Reasons`, `Tips`, `Tippers`, `tip finder`
 		/// # </weight>
-		#[weight = SimpleDispatchInfo::FixedNormal(50_000_000)]
-		fn close_tip(origin, hash: T::Hash) {
+		#[weight = 211_000_000 + 1_094_000 * MAX_TIPPERS_COUNT + T::DbWeight::get().reads_writes(3, 3)]
```

Comment:
I liked your suggestion to include the raw data with the weight paragraph. I also think it was suggested we choose "round" numbers for weights, but maybe we should double check with others.

I would write a weight of:

```
 #[weight = 200_000_000 + 1_000_000 * MAX_TIPPERS_COUNT + T::DbWeight::get().reads_writes(3, 3)] 
```

If I was doing this.

Reply:
it improves readibility but rounding 211 to 200 is 5% and 1_094 to 1_000 is 10%. I feel like its fine to just put the direct value with useless precision.
Maybe I can round a bit but not so much

Reply:
I kept 2 significative digits for each and rounded up to get them 
https://en.wikipedia.org/wiki/Significant_figures